diff --git a/node_modules/yahoo-fantasy/YahooFantasy.mjs b/node_modules/yahoo-fantasy/YahooFantasy.mjs
index 636728e..8a1e170 100644
--- a/node_modules/yahoo-fantasy/YahooFantasy.mjs
+++ b/node_modules/yahoo-fantasy/YahooFantasy.mjs
@@ -284,23 +284,27 @@ class YahooFantasy {
           });
 
           resp.on("end", () => {
-            data = JSON.parse(data);
-
-            if (data.error) {
-              if (/"token_expired"/i.test(data.error.description)) {
-                return this.refreshToken((err, data) => {
-                  if (err) {
-                    return reject(err);
-                  }
-
-                  return resolve(this.api(method, url, postData));
-                });
-              } else {
-                return reject(data.error);
+            try {
+              data = JSON.parse(data);
+
+              if (data.error) {
+                if (/"token_expired"/i.test(data.error.description)) {
+                  return this.refreshToken((err, data) => {
+                    if (err) {
+                      return reject(err);
+                    }
+
+                    return resolve(this.api(method, url, postData));
+                  });
+                } else {
+                  return reject(data.error);
+                }
               }
-            }
 
-            return resolve(data);
+              return resolve(data);
+            } catch(err) {
+              return reject(err)
+            }
           });
         })
         .on("error", (err) => {
